### PR TITLE
release-4.0.0 (HDS-2069) Select component (part I)

### LINKED ISSUE #376: Combobox: removing selected option and/or displaying all options if selection exists
## To which feature the request relates to

Seeing the full list of options after selection and/or removing selected option is difficult and rather unintuitive with Default single Combobox. After selection, the input text has to be either cleared with keyboard by selecting all and deleting, or with mouse by selecting the input text and clicking `cut` from the mouse actions menu to reset Combobox. I would say the current implementation is questionable from the accessibility point of view, too.

This problem can be produced in [Storybook](https://core-storybook.asti-dev.kaskotoimii.fi/?path=/story/components-combobox--default) by selecting e.g. "Valinta 2" and then opening the dropdown again where only "Valinta 2" with checkmark is displayed with no apparent cue of how to unselect the option, or how to see other options that should be available. Screenshot below.

## What
There's a few options to solve this. In my opinion best solution would be to combine `Select` and `Combobox` functionalities so that if Combobox is opened, even if something is already selected, the list of other options is always shown ([Select example in Storybook](https://core-storybook.asti-dev.kaskotoimii.fi/?path=/story/components-select--default)). Example of similar feature can be seen with [React Select's single select](https://react-select.com/home).

Alternative option (or even better: in addition to the previous solution) would be to implement `clearable` prop to default Combobox as currently only multiselect Combobox supports it. This could be done whilst implementing `clearable` in `Select` as described in the [other issue thread](https://github.com/City-of-Helsinki/helsinki-design-system/issues/358).

## Why

Current behaviour is confusing to the user and has a high probability that the user gets stuck. And in any case it creates a situation where unnecessary actions are needed if a selection needs to be changed afterwards.

## Related material

Screenshot for the problematic list view: Screenshot 2021-02-18 at 10 01 22

React Select example: https://react-select.com/home

### PR DESCRIPTION
**Pull request by NikoHelle**
**Note: target branch is [another select component branch with its own PR](https://github.com/City-of-Helsinki/helsinki-design-system/pull/1229).**

## Description

First commits for the Select component. Skeleton version with only single select and basic components.

## Related Issue

https://helsinkisolutionoffice.atlassian.net/browse/HDS-2069

## Motivation and Context

Making multiple PRs for the Select instead of one huge.

## How Has This Been Tested?

Unit tests

[Demo](https://city-of-helsinki.github.io/hds-demo/hds-2069-select-part-i/?path=/story/components-select--example)

## Add to changelog
Not until whole Select is ready

### CODE REVIEW

**Review by laurakarhu — APPROVED** [review]
Good job, couple of questions and remarks. Please have developer reviews as well :D

File: packages/react/src/components/select/Select.module.scss [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    outline: none;
+    padding: var(--spacing-s);
+    padding-right: var(--spacing-2-xs);
+
+    .labels {
+      box-sizing: border-box;
+      display: flex;
+      flex-flow: wrap;
+      flex-grow: 1;
+      max-height: 27px;
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/select/Select.module.scss:56]: It's body elements line-height. Changed to var(--line-height-l)

File: packages/react/src/components/select/Select.test.tsx
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/select/Select.test.tsx:1]: Jest cannot test css based stuff like widths, heights, overflows, line-breaks. We need Playwright for those.

File: packages/react/src/components/select/components/selectedOptions/ButtonWithSelectedOptions.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    'aria-controls': elementIds.selectionsAndListsContainer,
+    'aria-expanded': open,
+    'aria-haspopup': 'listbox',
+  };
+};
+
+/*
+This function updates the indicator showing how many options are hidden.
+Because the number of hidden elements can be n+1 digits, the space the indicator needs to vary.
+If the hidden element count is +9 before hidden elements are calculated it bay change to +10, the count requires more space.
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/select/components/selectedOptions/ButtonWithSelectedOptions.tsx:45]: Fixed.

File: packages/react/src/components/select/Select.module.scss
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/select/Select.module.scss:1]: The old one has none. We can discuss this later. The dropdown could be placed also in core. The select itself does not work without JS, so most of the css could not be demoed with stories.

File: packages/react/src/components/select/Select.module.scss [hunk 3]
```diff
@@ -0,0 +1,267 @@
+@import '../../styles/common.scss';
+@import '../dropdown/dropdown.common.scss';
+
+.root {
+  @extend %dropdownRoot;
+
+  display: flex;
+  flex-direction: column;
+  max-width: 420px;
```
[the remark this answers is not in the document]
Comment (NikoHelle) [reply packages/react/src/components/select/Select.module.scss:9]: Good point, but I can't remember the reason anymore. I believe this was defined in another Miro-board about Select fixes, but can't find it and Slack has no history that far so cannot find it from there either. Should be discussed and fixed after holidays.

**Pull request merged**